### PR TITLE
feat: add toggle ability

### PR DESCRIPTION
**Pull request by peldas**
Added the ability to toggle the WebSocket connection so that users can "hack" the text hooker page to pause/play the timer whenever mpv's pause status is toggled.

### CODE REVIEW

**Comment by kuroahna** [comment]
Thanks for the PR!

I wonder if we should actually generalize this problem. People might want to listen to other events provided by mpv and also potentially control mpv through the websocket. I kind of envision mpv_websocket to be a simple websocket proxy for mpv's JSON-IPC and let the client actually handle any application logic

We currently open a websocket on the default port 6677 and stream the subtitle text to all connected clients. We can open a 2nd websocket on port 6678 and simply proxy any JSON message coming in and out of the websocket.

CC: @Renji-XD What do you think?

Regardless, I think toggling the websocket connection is useful, so we can still merge this

File: mpv/scripts/run_websocket_server_linux.lua [hunk 1]
```diff
@@ -6,17 +6,37 @@ local utils = require 'mp.utils'
...
-    args = {
-        mpv_websocket_path,
-        "-m",
-        "/tmp/mpv-socket",
-        "-w",
-        "6677",
-    },
-})
+local function start_websocket()
+	mp.command_native_async({
```
Comment (kuroahna) [inline mpv/scripts/run_websocket_server_linux.lua:12]:
```suggestion
	initialised_websocket = mp.command_native_async({
```

File: mpv/scripts/run_websocket_server_windows.lua [hunk 2]
```diff
@@ -6,17 +6,39 @@ local utils = require 'mp.utils'
...
+
+local function end_websocket()
+	mp.abort_async_command(initialised_websocket)
+	initialised_websocket = nil
+end
+
+local function toggle_websocket()
+	local paused = mp.get_property_bool('pause')
+	if initialised_websocket and paused then
+		mp.osd_message('end websocket')
```
Comment (kuroahna) [inline mpv/scripts/run_websocket_server_windows.lua:35]: @peldas I assume this displays this message on mpv. Do we want to propagate this to the linux/mac script as well? Or should we delete this?

**Comment by Renji-XD** [comment]
> 
> CC: @Renji-XD What do you think?
> 

it is a valid use case but for me out of scope for my project. If you'll extend it to all messages on a 2nd port i'll may look into it again in the context of a new one in case i want to create a mpv browser control

**Comment by kuroahna** [comment]
> > CC: @Renji-XD What do you think?
> 
> it is a valid use case but for me out of scope for my project. If you'll extend it to all messages on a 2nd port i'll may look into it again in the context of a new one in case i want to create a mpv browser control page or similar in the future

Yep sounds good. Thanks for the response!

**Pull request merged**